### PR TITLE
Build: Fix slow tests

### PR DESCRIPTION
**Pull request by rozza**
JAVA-4145

https://spruce.mongodb.com/version/610bebf90ae606191fe957b7 - for the slow tests

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -274,10 +274,17 @@ configure(javaCodeCheckedProjects) {
...
     task testSlowOnly(type: Test) {
+        dependsOn('testSlowGroovy')
         useJUnitPlatform {
-            includeEngines('junit-jupiter', 'junit-vintage', 'spock')
+            includeEngines('junit-jupiter', 'junit-vintage')
             includeTags('Slow')
         }
+    }
+
+    task testSlowGroovy(type: Test) {
```
Comment (rozza) [inline build.gradle:284]: Had to create a custom type just for the groovy tests

File: build.gradle [hunk 2]
```diff
@@ -274,10 +274,17 @@ configure(javaCodeCheckedProjects) {
     }
 
     task testSlowOnly(type: Test) {
+        dependsOn('testSlowGroovy')
```
Comment (rozza) [inline build.gradle:277]: Add the spock test dependency

File: driver-core/src/test/functional/com/mongodb/internal/connection/AsyncStreamTimeoutsSpecification.groovy [hunk 3]
```diff
@@ -39,7 +40,6 @@ import static com.mongodb.ClusterFixture.getServerApi
 import static com.mongodb.ClusterFixture.getSslSettings
 import static com.mongodb.internal.connection.CommandHelper.executeCommand
 
-@IgnoreIf({ System.getProperty('ignoreSlowUnitTests') == 'true' })
```
Comment (rozza) [inline driver-core/src/test/functional/com/mongodb/internal/connection/AsyncStreamTimeoutsSpecification.groovy:42]: This seems to be a legacy system property

File: driver-core/src/test/functional/com/mongodb/internal/connection/AsyncStreamTimeoutsSpecification.groovy [hunk 4]
```diff
@@ -49,9 +49,9 @@ class AsyncStreamTimeoutsSpecification extends OperationFunctionalSpecification
     @IgnoreIf({ getSslSettings().isEnabled() })
     def 'should throw a MongoSocketOpenException when the AsynchronousSocket Stream fails to open'() {
         given:
-        def connection = new InternalStreamConnectionFactory(
+        def connection = new InternalStreamConnectionFactory(ClusterConnectionMode.SINGLE,
```
Comment (rozza) [inline driver-core/src/test/functional/com/mongodb/internal/connection/AsyncStreamTimeoutsSpecification.groovy:52]: These changes are just to bring the code up to date.

File: driver-core/src/test/unit/com/mongodb/internal/connection/DefaultConnectionPoolSpecification.groovy [hunk 5]
```diff
@@ -156,7 +156,7 @@ class DefaultConnectionPoolSpecification extends Specification {
         Thread.sleep(500)
 
         then: 'it prunes the existing connections and again ensures the minimum size of the pool'
-        connectionFactory.createdConnections.size() == 10
+        connectionFactory.createdConnections.size() == 5
```
Comment (rozza) [inline driver-core/src/test/unit/com/mongodb/internal/connection/DefaultConnectionPoolSpecification.groovy:159]: This was a test bug.

File: driver-sync/src/test/functional/com/mongodb/client/ClientSideEncryptionBsonSizeLimitsSpecification.groovy [hunk 6]
```diff
@@ -39,10 +39,10 @@ import static util.JsonPoweredTestHelper.getTestDocument
 
 class ClientSideEncryptionBsonSizeLimitsSpecification extends FunctionalSpecification {
 
-    private final String collectionName = 'ClientSideEncryptionBsonSizeLimitsSpecification'
+    private final String collName = 'ClientSideEncryptionBsonSizeLimitsSpecification'
```
Comment (rozza) [inline driver-sync/src/test/functional/com/mongodb/client/ClientSideEncryptionBsonSizeLimitsSpecification.groovy:42]: Side effect of Spock dependency update? Theres a `getCollectionName` method in the `FunctionalSpecification` and spock complained they clashed.

File: build.gradle (hunk 1 above)
Comment (stIncMale) [reply build.gradle:284]: Yes, thats's what seem to be suggested as the solution, which is unfortunate: https://github.com/spockframework/spock/issues/1288#issuecomment-791901698.

**Review by stIncMale — APPROVED** [review]
This is great, thank you @rozza! 💪

**Review by jyemin — APPROVED** [review]
LGTM

**Pull request merged**